stdlib/builtins.pyi

```diff
@@ -364,10 +364,24 @@ class float:
     def __bool__(self) -> bool: ...
 
 class complex:
-    @overload
-    def __new__(cls: type[Self], real: float = ..., imag: float = ...) -> Self: ...
-    @overload
-    def __new__(cls: type[Self], real: str | SupportsComplex | SupportsIndex | complex) -> Self: ...
+    if sys.version_info >= (3, 8):
+        # Python doesn't currently accept SupportsComplex for the second argument
+        @overload
+        def __new__(
+            cls: type[Self],
+            real: complex | SupportsComplex | SupportsFloat | SupportsIndex = ...,
```

Comment:
I find it weird that `SupportsComplex` doesn't include `complex`.

Reply:
You're not the only one — I believe this specific oddity was why they added `complex.__complex__` in 3.11 :)